airbyte-cdk/python/airbyte_cdk/sources/file_based/stream/default_file_based_stream.py

```diff
@@ -0,0 +1,130 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+import asyncio
+import logging
+from datetime import datetime
+from functools import cache
+from typing import Any, Iterable, List, Mapping, Optional, Union
+
+from airbyte_cdk.models import SyncMode
+from airbyte_cdk.sources.declarative.types import StreamSlice, StreamState
+from airbyte_cdk.sources.file_based.exceptions import MissingSchemaError, RecordParseError, SchemaInferenceError
+from airbyte_cdk.sources.file_based.remote_file import RemoteFile
+from airbyte_cdk.sources.file_based.schema_helpers import merge_schemas, type_mapping_to_jsonschema
+from airbyte_cdk.sources.file_based.schema_validation_policies import record_passes_validation_policy
+from airbyte_cdk.sources.file_based.stream import AbstractFileBasedStream
+from airbyte_cdk.sources.utils.record_helper import stream_data_to_airbyte_message
+
+
+class DefaultFileBasedStream(AbstractFileBasedStream):
+    """
+    The default file-based stream.
+    """
+
+    @property
+    def primary_key(self) -> Optional[Union[str, List[str], List[List[str]]]]:
+        return self.config.primary_key
+
+    def read_records(
+        self,
+        sync_mode: SyncMode,
+        cursor_field: List[str] = None,
+        stream_slice: Optional[StreamSlice] = None,
+        stream_state: Optional[StreamState] = None,
+    ) -> Iterable[Mapping[str, Any]]:
+        """
+        Yield all records from all remote files in `list_files_for_this_sync`.
+        """
+        schema = self._catalog_schema
+        if schema is None:
+            # On read requests we should always have the catalog available
+            raise MissingSchemaError("Expected `json_schema` in the configured catalog but it is missing.")
+        parser = self.get_parser(self.config.file_type)
+        for file in self.list_files_for_this_sync(stream_slice):
```

Comment:
nit: I think we should compute the files to read from the `stream_slices` method and store them in the slice. I can update the `list_files_for_this_sync` to reflect this as I implement incremental syncs